presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -104,6 +110,30 @@ private PlanNodeStatsEstimate filterForFalseExpression()
             return falseStatsBuilder.setOutputRowCount(0.0).build();
         }
 
+        @Override
+        protected PlanNodeStatsEstimate visitNotExpression(NotExpression node, Void context)
+        {
+            return differenceInStats(input, process(node.getValue()));
+        }
+
+        @Override
+        protected PlanNodeStatsEstimate visitLogicalBinaryExpression(LogicalBinaryExpression node, Void context)
+        {
+            PlanNodeStatsEstimate leftStats = process(node.getLeft());
+            PlanNodeStatsEstimate andStats = new FilterExpressionStatsCalculatingVisitor(leftStats, session, types).process(node.getRight());
+
+            switch (node.getType()) {
+                case AND:
+                    return andStats;
+                case OR:
+                    PlanNodeStatsEstimate rightStats = process(node.getRight());
+                    PlanNodeStatsEstimate sumStats = addStatsAndSumDistinctValues(leftStats, rightStats);
+                    return differenceInNonRangeStats(sumStats, andStats);
+                default:
+                    throw new IllegalStateException(format("Unimplemented logical binary operator expression %s", node.getType()));
```

Comment:
This is different case. As soon as we introduce logical binary expression other than AND or OR, this code needs to be updated -- e.g. "TODO add support for XOR" needs to be added. Otherwise we will never realize this is needed.